adapters/nexx360/nexx360.go

```diff
@@ -0,0 +1,246 @@
+	Placement string `json:"placement,omitempty"`
+}
+
+type Nexx360Caller struct {
+	Name    string `json:"name,omitempty"`
+	Version string `json:"version,omitempty"`
+}
+
+type ReqExt struct {
+	Nexx360 *ReqNexx360Ext `json:"nexx360,omitempty"`
+}
+
+type ReqNexx360Ext struct {
+	Caller []Nexx360Caller `json:"caller,omitempty"`
+}
+
+type Nexx360ResBidExt struct {
+	BidType string `json:"bidType,omitempty"`
+}
+
+// CALLER Info used to track Prebid Server
+// as one of the hops in the request to exchange
+var CALLER = Nexx360Caller{"Prebid-Server", "n/a"}
+
+func makeImps(impList []openrtb2.Imp) (MakeImpOutput, error) {
+	var output MakeImpOutput
+	var imps []openrtb2.Imp
+	for idx, imp := range impList {
+		var bidderExt adapters.ExtImpBidder
+		if err := jsonutil.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			return MakeImpOutput{}, &errortypes.BadInput{
+				Message: err.Error(),
+			}
+		}
+
+		var nexx360Ext openrtb_ext.ExtImpNexx360
+		if err := jsonutil.Unmarshal(bidderExt.Bidder, &nexx360Ext); err != nil {
+			return MakeImpOutput{}, &errortypes.BadInput{
+				Message: err.Error(),
+			}
+		}
+
+		var impExt Ext
+		impExt.Nexx360.TagId = nexx360Ext.TagId
+		impExt.Nexx360.Placement = nexx360Ext.Placement
+
+		impExtJSON, err := json.Marshal(impExt)
+		if err != nil {
+			return MakeImpOutput{}, &errortypes.BadInput{
+				Message: err.Error(),
+			}
+		}
+
+		imp.Ext = impExtJSON
+		imps = append(imps, imp)
+		if idx == 0 {
+			output.TagId = nexx360Ext.TagId
+			output.Placement = nexx360Ext.Placement
+		}
+
+	}
+	output.Imp = imps
+	return output, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var makeImp, err = makeImps(request.Imp)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	request.Imp = makeImp.Imp
+
+	urlBuilder, err := url.Parse(a.endpoint)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	query := url.Values{}
+	if makeImp.Placement != "" {
+		query["placement"] = []string{makeImp.Placement}
+	}
+	if makeImp.TagId != "" {
+		query["tag_id"] = []string{makeImp.TagId}
+	}
+	urlBuilder.RawQuery = query.Encode()
+
+	uri := urlBuilder.String()
+
+	reqExt, err := makeReqExt(request)
+	if err != nil {
+		return nil, []error{err}
+	}
+	request.Ext = reqExt
+
+	// Last Step
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		return nil, []error{err}
+	}
+	fmt.Printf("Nexx360: Request JSON: %s\n", string(reqJSON))
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json")
+
+	adapter := &adapters.RequestData{
+		Method:  "POST",
+		Uri:     uri,
+		Body:    reqJSON,
+		Headers: headers,
+		ImpIDs:  openrtb_ext.GetImpIDs(request.Imp),
+	}
+
+	return []*adapters.RequestData{adapter}, nil
+}
+
+func makeReqExt(request *openrtb2.BidRequest) ([]byte, error) {
+	var reqExt ReqExt
+
+	if len(request.Ext) > 0 {
+		if err := jsonutil.Unmarshal(request.Ext, &reqExt); err != nil {
```

Comment:
Is there a possibility for a `"nexx360"` field to come inside the `req.ext` field? How would it look? Can you add a JSON test case to illustrate?

Reply:
Fixed, with test attached (exemplary/simple-banner.json)